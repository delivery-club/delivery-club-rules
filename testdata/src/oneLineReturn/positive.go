package oneLineReturn

func myFunc() string                     { return "" }
func myFuncWithArgs(s, ss string) string { return s + ss }

func warning1() string {
	if true {
		var s = "" // want ` suggestion: return \"\"`
		return s
	}
	if true {
		var s = myFunc() // want `suggestion: return myFunc()`
		return s
	}
	if true {
		var s = myFuncWithArgs("12", "34") // want `\Qsuggestion: return myFuncWithArgs("12", "34")`
		return s
	}

	s := "" // want ` suggestion: return \"\"`
	return s
}
