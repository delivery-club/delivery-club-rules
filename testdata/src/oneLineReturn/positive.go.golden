package oneLineReturn

func myFunc() string                     { return "" }
func myFuncWithArgs(s, ss string) string { return s + ss }

func warning1() string {
	if true {
		return ""
	}
	if true {
		return myFunc()
	}
	if true {
		return myFuncWithArgs("12", "34")
	}

	return ""
}
